packages/flutter_test/lib/src/matchers.dart

```diff
@@ -1702,7 +1703,9 @@ class _MatchesGoldenFile extends AsyncMatcher {
       imageFuture = _captureImage(elements.single);
     }
 
-    final Uri testNameUri = _getTestNameUri(key, version);
+    final Uri testNameUri = _testingWithSkiaGold()
```

Comment:
It's really unfortunate that Skia Gold is leaking out into the matcher. Any way to refactor this so the Matcher doesn't have to know? Maybe you can just provide the client with all the information and the client just decides how to build the testNameUri?